sdk/python/feast/infra/offline_stores/snowflake_source.py

```diff
@@ -279,20 +280,21 @@ def get_table_column_names_and_types(
                 else:
                     row["snowflake_type"] = "NUMBERwSCALE"
 
-            elif row["type_code"] in [5, 9, 12]:
+            elif row["type_code"] in {5, 9, 12}:
                 error = snowflake_unsupported_map[row["type_code"]]
                 raise NotImplementedError(
                     f"The following Snowflake Data Type is not supported: {error}"
                 )
-            elif row["type_code"] in [1, 2, 3, 4, 6, 7, 8, 10, 11, 13]:
+            elif row["type_code"] in {1, 2, 3, 4, 6, 7, 8, 10, 11, 13}:
                 row["snowflake_type"] = snowflake_type_code_map[row["type_code"]]
             else:
                 raise NotImplementedError(
                     f"The following Snowflake Column is not supported: {row['column_name']} (type_code: {row['type_code']})"
                 )
 
         return [
-            (column["column_name"], column["snowflake_type"]) for column in metadata
+            (str(column["column_name"]), str(column["snowflake_type"]))
```

Comment:
do we need to fix one in the PR scope?

Reply:
Yes, it's for Pydantic v2's type check. The return type is "Iterable[Tuple[str, str]]"